<a href="https://colab.research.google.com/github/Deinbi/Challenge_Data_Science_1/blob/main/AluraStoreLatam.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

### Importación de datos



In [77]:
import pandas as pd

url = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_1%20.csv"
url2 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_2.csv"
url3 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_3.csv"
url4 = "https://raw.githubusercontent.com/alura-es-cursos/challenge1-data-science-latam/refs/heads/main/base-de-datos-challenge1-latam/tienda_4.csv"

tienda1 = pd.read_csv(url)
tienda2 = pd.read_csv(url2)
tienda3 = pd.read_csv(url3)
tienda4 = pd.read_csv(url4)

print("Tienda1:")
display(tienda1.head())
print()
print(tienda1.dtypes)
print()
print(tienda1.shape)
print("Tienda2:")
display(tienda2.head())
print()
print(tienda2.dtypes)
print()
print(tienda2.shape)
print("Tienda3:")
display(tienda3.head())
print()
print(tienda3.dtypes)
print()
print(tienda3.shape)
print("Tienda4:")
display(tienda4.head())
print()
print(tienda4.dtypes)
print()
print(tienda4.shape)


Tienda1:


Producto Categoría del Producto    Precio  Costo de envío  \
0  Asistente virtual           Electrónicos  164300.0          6900.0   
1    Mesa de comedor                Muebles  192300.0          8400.0   
2      Juego de mesa               Juguetes  209600.0         15900.0   
3         Microondas      Electrodomésticos  757500.0         41000.0   
4   Silla de oficina                Muebles  335200.0         20200.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      16/01/2021      Pedro Gomez          Bogotá             4   
1      18/05/2022  Beatriz Morales        Medellín             1   
2      15/03/2021   Juan Fernandez       Cartagena             1   
3      03/05/2022   Juan Fernandez            Cali             4   
4      07/11/2020    Maria Alfonso        Medellín             5   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   8   4.60971 -74.08175  
1  Tarjeta de crédito                   4   6.25184 -75.56359  
2  Tarjeta de crédito                   1  10.39972 -75.51444  
3               Nequi                   1   3.43722 -76.52250  
4               Nequi                   1   6.25184 -75.56359


Producto                   object
Categoría del Producto     object
Precio                    float64
Costo de envío            float64
Fecha de Compra            object
Vendedor                   object
Lugar de Compra            object
Calificación                int64
Método de pago             object
Cantidad de cuotas          int64
lat                       float64
lon                       float64
dtype: object

(2359, 12)
Tienda2:


Producto  Categoría del Producto    Precio  Costo de envío  \
0  Silla de oficina                 Muebles  283600.0         15300.0   
1         Pandereta  Instrumentos musicales   76800.0          2300.0   
2         Cama king                 Muebles  443300.0         23800.0   
3         Cama king                 Muebles  906200.0         46400.0   
4  Silla de oficina                 Muebles  397800.0         19400.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      04/08/2021  Izabela de León        Medellín             1   
1      26/11/2020   Blanca Ramirez        Medellín             3   
2      22/10/2021  Izabela de León         Pereira             5   
3      04/05/2022    Maria Alfonso            Cali             5   
4      26/02/2022    Lucas Olivera          Bogotá             4   

       Método de pago  Cantidad de cuotas      lat       lon  
0               Nequi                   1  6.25184 -75.56359  
1  Tarjeta de crédito                   4  6.25184 -75.56359  
2  Tarjeta de crédito                  10  4.81333 -75.69611  
3  Tarjeta de crédito                  10  3.43722 -76.52250  
4  Tarjeta de crédito                   3  4.60971 -74.08175


Producto                   object
Categoría del Producto     object
Precio                    float64
Costo de envío            float64
Fecha de Compra            object
Vendedor                   object
Lugar de Compra            object
Calificación                int64
Método de pago             object
Cantidad de cuotas          int64
lat                       float64
lon                       float64
dtype: object

(2359, 12)
Tienda3:


Producto  Categoría del Producto     Precio  Costo de envío  \
0   Secadora de ropa       Electrodomésticos  1285100.0         66600.0   
1      Mesa de noche                 Muebles   290100.0         13600.0   
2  Guitarra acústica  Instrumentos musicales   303500.0         17400.0   
3  Balón de voleibol    Deportes y diversión    58800.0          3300.0   
4            Armario                 Muebles   521100.0         25900.0   

  Fecha de Compra        Vendedor Lugar de Compra  Calificación  \
0      17/12/2021   Camila Rivera     Santa Marta             4   
1      10/02/2021   Maria Alfonso            Cali             3   
2      22/01/2023   Lucas Olivera          Bogotá             5   
3      22/03/2022     Pedro Gomez    Barranquilla             4   
4      12/01/2022  Blanca Ramirez          Bogotá             1   

       Método de pago  Cantidad de cuotas       lat       lon  
0  Tarjeta de crédito                   4  11.24079 -74.19904  
1  Tarjeta de crédito                   1   3.43722 -76.52250  
2  Tarjeta de crédito                   3   4.60971 -74.08175  
3  Tarjeta de crédito                   6  10.96854 -74.78132  
4  Tarjeta de crédito                   1   4.60971 -74.08175


Producto                   object
Categoría del Producto     object
Precio                    float64
Costo de envío            float64
Fecha de Compra            object
Vendedor                   object
Lugar de Compra            object
Calificación                int64
Método de pago             object
Cantidad de cuotas          int64
lat                       float64
lon                       float64
dtype: object

(2359, 12)
Tienda4:


Producto Categoría del Producto     Precio  Costo de envío  \
0         Lavavajillas      Electrodomésticos  1084800.0         56000.0   
1             Cama box                Muebles   473800.0         25400.0   
2  Balón de baloncesto   Deportes y diversión    23700.0             0.0   
3  Modelado predictivo                 Libros    57700.0          1300.0   
4          Auriculares           Electrónicos   138900.0          5600.0   

  Fecha de Compra         Vendedor Lugar de Compra  Calificación  \
0      11/11/2022    Rafael Acosta          Bogotá             4   
1      03/01/2020  Izabela de León          Cúcuta             5   
2      25/12/2022    Maria Alfonso        Medellín             5   
3      23/11/2021    Lucas Olivera        Medellín             3   
4      25/04/2022   Blanca Ramirez          Bogotá             4   

       Método de pago  Cantidad de cuotas      lat       lon  
0  Tarjeta de crédito                   3  4.60971 -74.08175  
1  Tarjeta de crédito                   2  7.89391 -72.50782  
2               Nequi                   1  6.25184 -75.56359  
3               Nequi                   1  6.25184 -75.56359  
4  Tarjeta de crédito                   7  4.60971 -74.08175


Producto                   object
Categoría del Producto     object
Precio                    float64
Costo de envío            float64
Fecha de Compra            object
Vendedor                   object
Lugar de Compra            object
Calificación                int64
Método de pago             object
Cantidad de cuotas          int64
lat                       float64
lon                       float64
dtype: object

(2358, 12)


De la importacion de datos, podemos notar que la Tienda 4, tiene un registro menos que el resto de las tiendas. Esto se puede deber a que en el mismo periodo de tiempo, se facturo menos o tambien, se puede deber a un error en los datos suministrados. De cualquier manera, esto tiene que constar en el informe final.

In [ ]:
print(tienda1.shape)
print(tienda2.shape)
print(tienda3.shape)
print(tienda4.shape)

#1. Análisis de facturación



In [5]:
total_tienda1 = tienda1['Precio'].sum()
total_tienda2 = tienda2['Precio'].sum()
total_tienda3 = tienda3['Precio'].sum()
total_tienda4 = tienda4['Precio'].sum()
print(f"La tienda 1, facturo: {total_tienda1}$")
print(f"La tienda 2, facturo: {total_tienda2}$")
print(f"La tienda 3, facturo: {total_tienda3}$")
print(f"La tienda 4, facturo: {total_tienda4}$")

La tienda 1, facturo: 1150880400.0$
La tienda 2, facturo: 1116343500.0$
La tienda 3, facturo: 1098019600.0$
La tienda 4, facturo: 1038375700.0$


Me interesa saber el valor promedio facturado por articulo, para relacionarlo con el enfoque comercial de cada tienda.

In [80]:
print(f"La tienda 1, facturo: {total_tienda1/tienda1.shape[0]}$")
print(f"La tienda 2, facturo: {total_tienda2/tienda2.shape[0]}$")
print(f"La tienda 3, facturo: {total_tienda3/tienda3.shape[0]}$")
print(f"La tienda 4, facturo: {total_tienda4/tienda4.shape[0]}$")



La tienda 1, facturo: 487867.9101314116$
La tienda 2, facturo: 473227.42687579483$
La tienda 3, facturo: 465459.7710894447$
La tienda 4, facturo: 440362.89228159457$


La tienda 1, claramente se orienta a articulos de mayor valor, esto se puede deber a que los venden mas caros o que tienen un marketing mas marcado a articulos de mayor valor. Mientras que la tienda 4, facturo solo un articulo menos y su promedio de valor por venta es menor. Esto me da la idea de que esta orientada a articulos de menor valor.

Si contabilizamos el costo de envio y se lo restamos a lo facturado (suponiendo que el envio esta dentro de la factura y el mismo, no esta generando ganacia) nos da el siguiente resultado:

In [6]:
print(f"La tienda 1, le queda limpio: {total_tienda1- tienda1['Costo de envío'].sum()}$")
print(f"La tienda 2, le queda limpio: {total_tienda2- tienda2['Costo de envío'].sum()}$")
print(f"La tienda 3, le queda limpio: {total_tienda3- tienda3['Costo de envío'].sum()}$")
print(f"La tienda 4, le queda limpio: {total_tienda4- tienda4['Costo de envío'].sum()}$")

La tienda 1, le queda limpio: 1089502500.0$
La tienda 2, le queda limpio: 1056858400.0$
La tienda 3, le queda limpio: 1039503000.0$
La tienda 4, le queda limpio: 983058300.0$


Por el contrario, si el envio se factura aparte y genera una ganancia:


In [46]:
print(f"La tienda 1, le queda limpio: {total_tienda1+ tienda1['Costo de envío'].sum()}$")
print(f"La tienda 2, le queda limpio: {total_tienda2+ tienda2['Costo de envío'].sum()}$")
print(f"La tienda 3, le queda limpio: {total_tienda3+ tienda3['Costo de envío'].sum()}$")
print(f"La tienda 4, le queda limpio: {total_tienda4+ tienda4['Costo de envío'].sum()}$")


La tienda 1, le queda limpio: 1212258300.0$
La tienda 2, le queda limpio: 1175828600.0$
La tienda 3, le queda limpio: 1156536200.0$
La tienda 4, le queda limpio: 1093693100.0$


### Como resultado del primer análisis, podemos ver que de cualquier manera, el peor de los casos es cuanto a ganancia y facturación, es la tienda 4.


# 2. Ventas por categoría

In [52]:
print(f"La tienda 1, ventas por {tienda1['Categoría del Producto'].value_counts()}")
print()
print(f"La tienda 2, ventas por {tienda2['Categoría del Producto'].value_counts()}")
print()
print(f"La tienda 3, ventas por {tienda3['Categoría del Producto'].value_counts()}")
print()
print(f"La tienda 4, ventas por {tienda4['Categoría del Producto'].value_counts()}")
print()



La tienda 1, ventas por Categoría del Producto
Muebles                    465
Electrónicos               448
Juguetes                   324
Electrodomésticos          312
Deportes y diversión       284
Instrumentos musicales     182
Libros                     173
Artículos para el hogar    171
Name: count, dtype: int64

La tienda 2, ventas por Categoría del Producto
Muebles                    442
Electrónicos               422
Juguetes                   313
Electrodomésticos          305
Deportes y diversión       275
Instrumentos musicales     224
Libros                     197
Artículos para el hogar    181
Name: count, dtype: int64

La tienda 3, ventas por Categoría del Producto
Muebles                    499
Electrónicos               451
Juguetes                   315
Electrodomésticos          278
Deportes y diversión       277
Libros                     185
Instrumentos musicales     177
Artículos para el hogar    177
Name: count, dtype: int64

La tienda 4, ventas por Categoría 

Si sólo nos quedamos con lo más vandido de cada tienda:

In [ ]:
categoria_mas_vendida = tienda1['Categoría del Producto'].value_counts()
print("La categoria mas vendida en la tienda 1 es:")
display(categoria_mas_vendida.head(1))
print("")
categoria_mas_vendida = tienda2['Categoría del Producto'].value_counts()
print("La categoria mas vendida en la tienda 2 es:")
display(categoria_mas_vendida.head(1))
print("")
categoria_mas_vendida = tienda3['Categoría del Producto'].value_counts()
print("La categoria mas vendida en la tienda 3 es:")
display(categoria_mas_vendida.head(1))
print("")
categoria_mas_vendida = tienda4['Categoría del Producto'].value_counts()
print("La categoria mas vendida en la tienda 4 es:")
display(categoria_mas_vendida.head(1))
print("")


## Ahora, desde una evaluacion mas general de las categorias, podriamos hacer la siguiente relacion:

### Top categorías en todas las tiendas:

*Muebles y Electrónicos* son consistentemente las categorías más vendidas. Esto puede deberse a alta demanda, promociones frecuentes o una mejor presentación en tienda.

*Juguetes* mantiene una posición intermedia-alta, especialmente fuerte en la **Tienda 4** (338 unidades).

*Electrodomésticos y Deportes y diversión* están en el medio, pero tienden a ser menos consistentes: Ej: **Tienda 3** tiene menos ventas de *Electrodomésticos* que las demás.

*Libros, Instrumentos Musicales y Artículos para el hogar* , están en el fondo del ranking: Aunque hay una ligera diferencia, como en **Tienda 2** donde *Instrumentos Musicales* vendió relativamente más.

### Variaciones destacables:

*Instrumentos Musicales* tuvo un mejor desempeño en **Tienda 2**.

*Artículos para el hogar* tuvo más ventas en la **Tienda 4** (posible enfoque o promoción local).

**Tienda 3** tuvo un pico en ventas de *Muebles*.

# 3. Calificación promedio de la tienda


In [76]:
print(f"La tienda 1, tienen una calificacion promedio {tienda1['Calificación'].mean()}")
print()
print(f"La tienda 2, tienen una calificacion promedio {tienda2['Calificación'].mean()}")
print()
print(f"La tienda 3, tienen una calificacion promedio {tienda3['Calificación'].mean()}")
print()
print(f"La tienda 4, tienen una calificacion promedio {tienda4['Calificación'].mean()}")
print()


La tienda 1, tienen una calificacion promedio 3.976685036032217.2f

La tienda 2, tienen una calificacion promedio 4.037303942348453

La tienda 3, tienen una calificacion promedio 4.048325561678677

La tienda 4, tienen una calificacion promedio 3.9957591178965224



# Análisis Combinado: Ventas + Satisfacción

## Tienda 3 es la mejor balanceada:

Tiene la calificación promedio más alta (4.05) y el segundo mayor volumen de ventas.

*Esto sugiere una muy buena gestión general: procesos eficientes, atención al cliente y surtido atractivo.*

## Tienda 2 también destaca:

Tiene la mayor cantidad de ventas y una satisfacción elevada (4.04).

Puede ser el mejor caso de éxito comercial.

## Tienda 4 tiene buen volumen de ventas pero satisfacción más baja (aunque sigue siendo decente).

Podría investigarse si el crecimiento en ventas está presionando la experiencia del cliente (tiempos de espera, disponibilidad, trato, etc.).

## Tienda 1 tiene el menor nivel de ventas y la satisfacción más baja. **Es la más preocupante en términos de desempeño general.**

Puede requerir una revisión integral: atención, logística, surtido, ambiente en tienda, o capacitación del personal.

# Resultados relacionados con la facturacion:

**Tienda 1** lidera en facturación
Aunque no tiene la mejor calificación, su precio promedio por unidad es el más alto.
Esto sugiere que vende productos de mayor valor agregado, posiblemente más muebles o electrónicos de gama alta.

*Pero cuidado: la satisfacción más baja puede indicar que los clientes no perciben un valor equivalente al precio.*

**Tienda 2 y Tienda 3** tienen buen equilibrio
Tienda 2 mantiene una buena facturación y excelente satisfacción.
Tienda 3 tiene la mejor satisfacción y facturación razonablemente alta, con el precio promedio más competitivo.

*Estas tiendas pueden estar aplicando una estrategia de valor y servicio, clave para fidelizar.*

Tienda 4 vende más en unidades, pero factura menos
Tiene el precio por unidad más bajo y menor facturación, a pesar de tener el mayor número de unidades vendidas.
Indica posible enfoque en productos económicos o descuentos agresivos. Tiene una satisfacción decente, pero no tan destacada para justificar una estrategia de volumen puro.

# 4. Productos más y menos vendidos

In [45]:
print(f"La tienda 1, tienen como producto mas vendido: {tienda1['Producto'].value_counts().idxmax()} y menos vendido: {tienda1['Producto'].value_counts().idxmin()}")
print()
print(f"La tienda 2, tienen como producto mas vendido: {tienda2['Producto'].value_counts().idxmax()} y menos vendido: {tienda2['Producto'].value_counts().idxmin()}")
print()
print(f"La tienda 3, tienen como producto mas vendido: {tienda3['Producto'].value_counts().idxmax()} y menos vendido: {tienda3['Producto'].value_counts().idxmin()}")
print()
print(f"La tienda 4, tienen como producto mas vendido: {tienda4['Producto'].value_counts().idxmax()} y menos vendido: {tienda4['Producto'].value_counts().idxmin()}")
print()


La tienda 1, tienen como producto mas vendido: Microondas y menos vendido: Auriculares con micrófono

La tienda 2, tienen como producto mas vendido: Iniciando en programación y menos vendido: Juego de mesa

La tienda 3, tienen como producto mas vendido: Kit de bancas y menos vendido: Bloques de construcción

La tienda 4, tienen como producto mas vendido: Cama box y menos vendido: Guitarra eléctrica



# 5. Envío promedio por tienda

In [85]:
print(f"La tienda 1, tienen como costo de envio promedio: {tienda1['Costo de envío'].mean()}")
print()
print(f"La tienda 2, tienen como costo de envio promedio: {tienda2['Costo de envío'].mean()}")
print()
print(f"La tienda 3, tienen como costo de envio promedio: {tienda3['Costo de envío'].mean()}")
print()
print(f"La tienda 4, tienen como costo de envio promedio: {tienda4['Costo de envío'].mean()}")
print()

La tienda 1, tienen como costo de envio promedio: 26018.60958033065

La tienda 2, tienen como costo de envio promedio: 25216.235693090293

La tienda 3, tienen como costo de envio promedio: 24805.680373039424

La tienda 4, tienen como costo de envio promedio: 23459.457167090754



Costo de Envío Promedio por Tienda

Tienda	Costo Promedio de Envío	Posición (Menor a Mayor)

Tienda 4	23,459.46	🟢 Más bajo

Tienda 3	24,805.68	🟢 Segundo más bajo

Tienda 2	25,216.24	🟡 Medio

Tienda 1	26,018.61 🔴 Más alto